tools/osbuild-image-test

```diff
@@ -412,7 +391,72 @@ def clean_output(folder):
             shutil.rmtree(os.path.join(root, d))
 
 
-#pylint: disable=too-many-branches
+def run_tests_cases(args):
+    print(f"Running in {os.path.realpath(os.curdir)}")
+
+    tests = TestCases.load(
+        args.arch,
+        args.distro,
+        args.name,
+        args.verbose,
+        args.libdir,
+        OSBuild(args.osbuild, args.store, args.output, args.verbose),
+        ImageInfo(args.image_info, args.verbose),
+        args.db_path)
+
+    success = True
+    if not tests:
+        print(f"{RED}FAIL{RESET}: no tests to run")
+        success = False
+
+    for test in tests:
+        print(f"{test.id}", end=" ", flush=True)
+        # Skip for dry run
+        if args.dry_run:
+            print(f"Dry run: {GREEN}SUCCESS{RESET}")
+            continue
+        # Skip test cases that have failed validation of their manifest
+        if test.error:
+            print(f"{YELLOW}Warning (ignoring){RESET}: {test.error}")
+            continue
+
+        # If the generator_mode flag is set, the program behaves as an
+        # image-info generator and skip the validation step. Which means that it
+        # won't error out if the comparison fails. Also it will trigger the
+        # build for every image regardless of the presence of an image-info in
+        # the corresponding test case to validate against.
+        if args.generator_mode:
+            print("Building only:", end=" ", flush=True)
+        else:
+            print("Building & Comparing:", end=" ", flush=True)
+            # If the program is configured to build & compare, then if the
+            # test case haven't an image-info to compare to, it's useless to
+            # build it. Because in this mode we are only interested in comparing
+            # with what's in the DB.
+            if not test.desired_image_info:
+                print(f"{YELLOW}Warning (ignoring){RESET}: No image info to compare to")
+                continue
+
+        if test.build():
+            # Every generated image-info needs to be written to disk regardless
+            # of the program running in generated mode or not. So that by
+            # running the program on a CI, one can download an image-info that
+            # failed comparison.
+            test.dump(args.image_info_output)
+            if not args.generator_mode:
+                test.compare()
+        if test.error:
+            print(f"{RED}FAIL{RESET}: {test.error}")
+            success = False
+        else:
+            print(f"{GREEN}SUCCESS{RESET}")
+
+        # in any case, clean the output folder to avoid filling out disk space
+        clean_output(args.output)
+
+    return 1 if not success else 0
```

Comment:
nitpick: you could simplify this as `return 0 if success else 1` 😇

Reply:
ahah, indeed!